connectors-home/modules/ROOT/pages/introduction/intro-config-oauth2-cloudhub.adoc

```diff
@@ -0,0 +1,39 @@
+
+== How to Configure OAuth 2.0 for a Connector on CloudHub
+
+Configuring OAuth 2.0 for a connector on CloudHub is the same as configuring OAuth 2.0 for a connector locally, except with CloudHub you are behind an extra server. You must keep in mind the following when configuring OAuth 2.0 for a connector on CloudHub:
+
+* Connector Listener
++
+** CloudHub manages some xref:runtime-manager::cloudhub-manage-props.adoc#cloudhub-reserved-properties[properties] for you automatically, such as `${http.port}` and `${https.port}`. Use these properties when configuring your *HTTP Listener*.
++
+image::intro-config-oauth2-cloudhub-img1.png[Http Listener configuration window for CloudHub example]
++
+** When deployed on the cloud, the `localhost`, `0.0.0.0`, and `127.0.0.1` hosts are no longer publicly reachable. When configuring your OAuth provider, you must replace all links containing these hosts with the actual CloudHub provided public addresses. For CloudHub, this address is in the format `{ch-app-name}.{region}.cloudhub.io`. For example, an HTTP application named `ch-example-app` deployed in the Europe (London) xref:runtime-manager::cloudhub-networking-guide.adoc#regional-services[region] has public address of `\http://ch-example-app.uk-e1.cloudhub.io`.
+
+* OAuth Provider Setup
++
+** If you set *Callback Path* to `/callback`, your public callback URL is `{ch-app-name}.{region}.cloudhub.io/callback`. This is the callback URL you want to register with your OAuth provider as your redirect URI.
++
+image::intro-config-oauth2-cloudhub-idp-example.png[OKTA Example]
++
+This must also be configured in your connector as the *External Callback Url*. You can use the `${fullDomain}` property that CloudHub configures for you to automatically obtain the proper host name.
++
+image::intro-config-oauth2-cloudhub-callback-config.png[External Callback Url field configuration window]
++
+NOTE: If you are deploying to multiple regions or have some other gateway configured, you must use the host name of the load balancer or the gateway in your OAuth provider callback URL and in *External Callback Url*.
++
+In OAuth 2.0 configuration, the user (not the OAuth provider) reaches the callback endpoint from their browser. Ensure your region configuration is correct. If you create your CloudHub app in the US region and your users connect to the app from Europe, the request is blocked by the gateway.
+
+== Performing the OAuth Dance
+
+After OAuth 2.0 configuration is complete, you can deploy your application to CloudHub and initialize the OAuth dance by navigating to your public hostname on your authorization endpoint. For example, with the above configuration, you can navigate to `{ch-app-name}.{region}.cloudhub.io/authorize`. This starts the OAuth dance and prompts you to login with your OAuth provider.
```

Comment:
```suggestion
After you complete the OAuth 2.0 configuration, you can deploy your application to CloudHub and initialize the OAuth dance by navigating to your public hostname on your authorization endpoint. For example, with the above configuration, you can navigate to `{ch-app-name}.{region}.cloudhub.io/authorize`. This starts the OAuth dance and prompts you to log in with your OAuth provider.
```

Reply:
For accessibility reasons, we shouldn't use the word "above." Maybe "previous" would work instead. (Sometimes it is hard to reword things like this.)